invenio_sipstore/archivers/bagit_archiver.py

```diff
@@ -54,25 +63,137 @@ def get_all_files(self):
+        :type patch_of: invenio_sipstore.models.SIP or None
+        :type bool include_missing_files: If True and when 'patch_of' is used,
+            include the files that are missing in the SIP w.r.t. to
+            the 'patch_of' SIP in the manifest.
+            The opposite (include_missing_files=False) is equivalent to
+            treating those as explicitly deleted.
         :returns: a dictionnary with the filenames as keys, and size and
             checksum as value
         :rtype: dict
         """
+        sipfiles = self.sip.files
+        if patch_of:
+            fetch_info = {}
+            sipfiles = []  # We will pick sipfiles for writing manually
+            sip_manifest = self.bagit_metadata(patch_of)['manifest']
+
+            # Helper mappings
+            # File UUID to manifest item mapping (select only the files data,
+            # not metadata files).
+            id2mi = dict((v['file_uuid'], dict(filename=k, meta=v)) for k, v in
+                         sip_manifest.items() if 'file_uuid' in v)
+            # File UUID to SIP File mapping
+            id2sf = dict((str(file.file.id), file) for file in self.sip.files)
+
+            previous_files_s = set(id2mi.keys())
+            sip_files_s = set(id2sf.keys())
+            if include_missing_files:
+                fetched_uuids = previous_files_s
+            else:
+                fetched_uuids = previous_files_s & sip_files_s
+            stored_uuids = sip_files_s - previous_files_s
+
+            for uuid in fetched_uuids:
+                if uuid in id2sf:
+                    filename = join(filesdir, id2sf[uuid].filepath)
+                else:
+                    filename = id2mi[uuid]['filename']
+                fetch_info[filename] = id2mi[uuid]['meta']
+
+            for uuid in stored_uuids:
+                sipfiles.append(id2sf[uuid])
+
+        # Copy the files
         files_info = super(BagItArchiver, self).create(
-            filesdir=join('data', 'files'), metadatadir='data')
+            filesdir=filesdir, metadatadir=metadatadir, sipfiles=sipfiles)
+        # Add the files that will be fetched to the files_info,
+        # so they will be included in manifest
+        if patch_of:
+            files_info.update(fetch_info)
 
         self.autogenerate_tags(files_info)
-
         metadata_info = self._save_file(*self.get_manifest_file(files_info))
+        if patch_of and fetch_info:
+            # Write the fetch.txt file if any of the files are to be fetched
+            metadata_info.update(
+                self._save_file(*self.get_fetch_file(fetch_info)))
         metadata_info.update(self._save_file(*self.get_bagit_file()))
         metadata_info.update(self._save_file(*self.get_baginfo_file()))
         metadata_info.update(
             self._save_file(*self.get_tagmanifest(metadata_info)))
 
+        if create_bagit_metadata:
+            # Store what we archived in BagIt
+            bagit_meta = {}
```

Comment:
Added the schema https://github.com/inveniosoftware/invenio-sipstore/pull/30/files#diff-a56fed499f498717abb66ba8c12807f9